camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/creation/BrooklynEntityDecorationResolver.java

```diff
@@ -227,7 +234,54 @@ protected Object getDecorationAttributeJsonValue(ConfigBag attrs) {
 
         @Override
         protected void addDecorationFromJsonMap(Map<?, ?> decorationJson, List<SpecParameter<?>> decorations) {
-            throw new IllegalStateException("Not called");
+            throw new UnsupportedOperationException("SpecParameterResolver.addDecorationFromJsonMap should never be called.");
+        }
+    }
+
+    public static class TagsResolver extends BrooklynEntityDecorationResolver<Iterable<Object>> {
+        protected TagsResolver(BrooklynYamlTypeInstantiator.Factory instantiator) {
+            super(instantiator);
+        }
+
+        @Override
+        public void decorate(EntitySpec<?> entitySpec, ConfigBag attrs, Set<String> encounteredRegisteredTypeIds) {
+            if (getDecorationAttributeJsonValue(attrs) != null) {
+                entitySpec.tagsAdd((Iterable<Object>)getDecorationAttributeJsonValue(attrs));
+            }
+        }
+
+        @Override
+        protected String getDecorationKind() {
+            return "Brooklyn Tags";
+        }
+
+        @Override
+        protected Object getDecorationAttributeJsonValue(ConfigBag attrs) {
```

Comment:
You can declare the return type as `Iterable<Object>`, and therefore avoid the cast in `decorate()`.

As an aside, the `entitySpec.tagsAdd() should really take an `Iterable<?>` rather than an `Iterable<Object>`, but don't worry about that for this PR.